dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+            installation_statuses = []
+            # default to orphan package in case of install from .ndjson
+            manifest = PackageManifest(OrphanPackageData)
+            package = Package(manifest)
+            print(filepath)
+            if filetype == 'application/x-tar' or encoding == 'gzip':
+                # handle tarfile
+                tar = tarfile.open(filepath)
+                try:
+                    manifest = tar.extractfile('manifest.json')
+                    manifest = PackageManifest(manifest.read().decode('utf8'))
+                    package = Package(manifest)
+                    package.create_packages_index()
+                    # check if package exists already.
+                    existing = Package.find_by_slug(package.manifest.create_slug())
+                    if existing:
+                        rmexisting = input(f"A Package titled {existing.manifest.name} is already installed, do you want to uninstall it? [y/n]") in "yY"
+                        if rmexisting:
+                            rmsuccess = existing.uninstall(self.kibana_url, auth=(username, password), force=True)
+                            if rmsuccess:
+                                print(f"Successfully removed existing package {existing.manifest.name}.")
+                    
+                except KeyError:
+                    print("Package must contain a manifest.json")
+                    exit(0)
+                for member in manifest.file_list:
+                    # should we validate the json before sending it up to kibana?
+                    kibana_objects_file = BytesIO(tar.extractfile(member).read())
+                    result = self.import_kibana_saved_objects(username=username, password=password,
+                                                    kibana_objects_file=kibana_objects_file)
+                    kibana_objects_file.close()
+                    installation_statuses.append(self._process_package_installation_results(package, result))
+            # Should we remove this and ONLY install validatable packages?
+            elif filetype in ('application/json', 'text/plain') and filepath.endswith('ndjson'):
+                with open(filepath, 'r') as kibana_objects_file:
+                    result = self.import_kibana_saved_objects(username=username, password=password,
+                                                    kibana_objects_file=kibana_objects_file)
+                    installation_statuses.append(self._process_package_installation_results(package, result))
+            else:
+                self.logger.error("Files must be one of: .ndjson, .json, .tar.xz, .tar.gz")
+                # TODO raise exception
+        
+        if not all(installation_statuses):
+            print(f"\r\n{package.manifest.name} installation failed.")
+            if not self.verbose:
+                print("Use --verbose flag to see more error detail.\n")
+        else:
+            package.register()
```

Comment:
`package` is referenced potentially before assignment (if `filelist` is empty)

Reply:
filelist is never empty, it is validated as a list with at least one item from the schema.